plugins/Ecommerce/Controller.php

```diff
@@ -44,7 +44,7 @@ public function getSparklines()
             $goalDefinition['name'] = $this->translator->translate('Goals_Ecommerce');
             $goalDefinition['allow_multiple'] = true;
         } else {
-            $goals = GoalsApi::getInstance()->getGoals($this->idSite);
+            $goals = Request::processRequest('Goals.getGoals', ['idSite' => $this->idSite]);
```

Comment:
do we need the `filter_limit = -1` here as well to be safe and set default to `[]`? eg the controller might send a filter_limit etc?